docs/flask.asciidoc

```diff
@@ -0,0 +1,232 @@
+
+NOTE: If you use Flask with uwsgi, make sure to
+http://uwsgi-docs.readthedocs.org/en/latest/Options.html#enable-threads[enable
+threads].
+
+[float]
+[[flask-setup]]
+=== Setup
+
+To set up the agent, you need to initialize it with appropriate settings.
+
+The settings are configured either via environment variables,
+the application's settings, or as initialization arguments.
+
+You can find a list of all available settings in the <<configuration, Configuration>> page.
+
+To initialize the agent for your application using environment variables:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+apm = ElasticAPM(app)
+----
+
+To configure the agent using `ELASTIC_APM` in your application's settings:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+
+app.config['ELASTIC_APM'] ={
+    'APP_NAME': '<APP-NAME>',
+    'SECRET_TOKEN': '<SECRET-TOKEN>',
+}
+apm = ElasticAPM(app)
+----
+
+The final option is to initialize the agent with the settings as arguments:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+
+apm = ElasticAPM(app, app_name='<APP-ID>', secret_token='<SECRET-TOKEN>')
+----
+
+[float]
+[[flask-debug-mode]]
+==== Debug Mode
+
+Please note that errors and transactions will only be sent to the apm server
+if your app is *not* in
+http://flask.pocoo.org/docs/0.12/quickstart/#debug-mode[debug mode].
+
+To force the agent to send data during development,
+set the value of `DEBUG` in the `ELASTIC_APM` dictionary to `True`:
+
+[source,python]
+----
+app.config['ELASTIC_APM'] = {
+        'APP_NAME': '<APP-NAME>',
+        'SECRET_TOKEN': '<SECRET-TOKEN>',
+        'DEBUG': True
+}
+----
+
+[float]
+[[flask-building-applications-on-the-fly]]
+==== Building applications on the fly?
+
+You can use the agent's `init_app` hook for adding the application on the fly:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+apm = ElasticAPM()
+
+def create_app():
+    app = Flask(__name__)
+    apm.init_app(app, app_name='<APP-NAME>', secret_token='<SECRET-TOKEN>')
+    return app
+----
+
+[float]
+[[flask-usage]]
+=== Usage
+
+Once you have configured the agent,
+it will automatically track transactions and capture uncaught exceptions within Flask.
+If you want to send additional events,
+a couple of shortcuts are provided on the ElasticAPM Flask middleware object
+by raising an exception or logging a generic message.
+
+Capture an arbitrary exception by calling `capture_exception`:
+
+[source,python]
+----
+try:
+    1 / 0
+except ZeroDivisionError:
+    apm.capture_exception()
+----
+
+Log a generic message with `capture_message`:
+
+[source,python]
+----
+apm.capture_message('hello, world!')
+----
+
+[float]
+[[flask-logging]]
+=== Logging
+
+Passing `logging=True` to the `ElasticAPM` constructor will make the agent
+automatically log errors that are logging using Python's built-in `logging` module.
```

Comment:
I would delete the `that are logging` part to have 

"Passing `logging=True` to the `ElasticAPM` constructor will make the agent automatically log errors, using Python's built-in `logging` module."

Reply:
The sentence definitely feels awkward right now, but your version isn't quite correct I think. Maybe

> Passing `logging=True` to the `ElasticAPM` constructor will make the agent automatically log all log messages from Python's built-in `logging` module.